src/model/stores/main/actions.ts

```diff
@@ -303,6 +328,11 @@ export function resumeLogin(
       user = mapOAuth2UserToUser(mapiUser);
       dispatch(loginSuccess(user));
 
+      const metadata = await auth.getImpersonationMetadata();
+      if (metadata) {
+        dispatch(setImpersonation(metadata));
```

Comment:
I'm not sure, but maybe we also need to handle impersonation in this block above?
```
// Try to resume GS user first.
user = fetchUserFromStorage();
if (user) {
      dispatch(loginSuccess(user));
      return Promise.resolve(user);
}
```